### PR TITLE
[postgresqlreceiver] bump supported PG version to 13

### LINKED ISSUE #30923: receiver/postgresql: bump PostgreSQL minimal supported version to latest official supported
### Component(s)

receiver/postgresql

### Is your feature request related to a problem? Please describe.

I see that this receiver is marked as supporting PG 9.6+ in the [README](https://github.com/collibra/opentelemetry-collector-contrib/blob/816671e3a2378b40c1d24114a09efc93b0c2e63d/receiver/postgresqlreceiver/README.md?plain=1#L24). Do we still want to keep this much backwards compatibility given that this is quite an old, and unsupported for 2+ years, version now ([list of PG versions here](https://www.postgresql.org/support/versioning/)). It seems that the latest supported version is PG 12 with still ~9 months of support and we should maybe align on this?
This can make improvements, using more recent features (a bit?) harder or use less maintained/out-of-date modules versions (e.g. https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30831)

### Describe the solution you'd like

Update the prerequisites requirements for using this receiver (+ integration tests) to the latest officially supported PostgreSQL version

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kevinnoel-be**
#### Description
Update the prerequisites requirements for using this receiver (+ integration tests) to the latest officially supported PostgreSQL version

#### Link to tracking issue
Fixes #30923

#### Testing
Unit & integration tests

#### Documentation
Updated README

### CODE REVIEW

File: receiver/postgresqlreceiver/integration_test.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func integrationTest(name string, databases []string) func(*testing.T) {
 		NewFactory(),
 		scraperinttest.WithContainerRequest(
 			testcontainers.ContainerRequest{
-				Image: "postgres:9.6.24",
+				Image: "postgres:12.20",
```
Comment (atoulme) [inline receiver/postgresqlreceiver/integration_test.go:55]: Can we support both versions?

Comment (kevinnoel-be) [reply receiver/postgresqlreceiver/integration_test.go:55]: In test or more generally? This is only moving the minimal requirement to the last supported version but doesn't actually change anything

Comment (MovieStoreGuy) [reply receiver/postgresqlreceiver/integration_test.go:55]: I think the question is more, does this mean that backwards compatibility is preserved?

File: receiver/postgresqlreceiver/README.md [hunk 2]
```diff
@@ -12,13 +12,13 @@
...
 
-This receiver queries the PostgreSQL [statistics collector](https://www.postgresql.org/docs/9.6/monitoring-stats.html).
+This receiver queries the PostgreSQL [statistics collector](https://www.postgresql.org/docs/12/monitoring-stats.html).
 
 > :construction: This receiver is in **BETA**. Configuration fields and metric data model are subject to change.
 
 ## Prerequisites
 
-This receiver supports PostgreSQL versions 9.6+
+This receiver supports PostgreSQL versions 12.20+
```
Comment (MovieStoreGuy) [inline receiver/postgresqlreceiver/README.md:21]: This should go into more details on the supported version, I think `12.20` is somewhat miss leading in the sense that we could be backwards compatible up to 12.20.

Comment (kevinnoel-be) [reply receiver/postgresqlreceiver/README.md:21]: Maybe we shouldn't really mention a version but only link to the latest supported upstream version then?

File: receiver/postgresqlreceiver/integration_test.go (hunk 1 above)
Comment (djaglowski) [reply receiver/postgresqlreceiver/integration_test.go:55]:
```suggestion
				Image: "postgres:13.18",
```

File: .chloggen/bump_minimal_pg_version.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: deprecation
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: postgresqlreceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Minimal supported PostgreSQL version updated from 9.6 to 12.0
```
Comment (djaglowski) [inline .chloggen/bump_minimal_pg_version.yaml:10]:
```suggestion
note: Minimal supported PostgreSQL version updated from 9.6 to 13.0
```

File: receiver/postgresqlreceiver/README.md [hunk 4]
```diff
@@ -12,13 +12,13 @@
 [contrib]: https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol-contrib
 <!-- end autogenerated section -->
 
-This receiver queries the PostgreSQL [statistics collector](https://www.postgresql.org/docs/9.6/monitoring-stats.html).
+This receiver queries the PostgreSQL [statistics collector](https://www.postgresql.org/docs/12/monitoring-stats.html).
```
Comment (djaglowski) [inline receiver/postgresqlreceiver/README.md:15]:
```suggestion
This receiver queries the PostgreSQL [statistics collector](https://www.postgresql.org/docs/13/monitoring-stats.html).
```

**Review by djaglowski — COMMENTED** [review]
I'm in support of updating the versions. Looks like 12 is no longer supported as of a few weeks ago.

File: receiver/postgresqlreceiver/README.md (hunk 2 above)
Comment (kevinnoel-be) [reply receiver/postgresqlreceiver/README.md]: Rephrased this sentence with a link to PG versions documentation

**Review by djaglowski — APPROVED** [review]

**Pull request merged**